### ISSUE TITLE
Regression in 2.14

### ISSUE THREAD

**Issue by bolducfr** [PyCQA/pylint#6803]
### Bug description

The following Python code contains a redefined variable (instrument) in the inner loop:
```python
def conflate(book):
    result = {}
    for instrument, trades in book.items():
        total = 0
        for qty, side1, side2, instrument in trades:
            if side1 != side2:
                total += qty
        result[instrument] = total
    return result
```

Pylint 2.13 correctly detects the error with the following message:
```
************* Module test
test.py:5:8: W0621: Redefining name 'instrument' from outer scope (line 3) (redefined-outer-name)
```

However, Pylint 2.14 fails to detect it.

### Command used

```shell
pytlint test.py
```

### Pylint output

```shell
Outputs nothing.
```

### Expected behavior

Output:
************* Module test
test.py:5:8: W0621: Redefining name 'instrument' from outer scope (line 3) (redefined-outer-name)

### Pylint version

```shell
2.14
```

**Comment by Pierre-Sassoulas** [comment]
The doc got better this release too : https://pylint.pycqa.org/en/latest/user_guide/messages/warning/redefined-loop-name.html?highlight=redefined-loop-name 😄

**Comment by Pierre-Sassoulas** [comment]
I think the issue is that you're using ``load-plugin`` instead of ``--load-plugins``. Don't you get a warning for this ?

**Comment by Pierre-Sassoulas** [comment]
Opened https://github.com/PyCQA/pylint/issues/6810 to address this

**Comment by bolducfr** [comment]
I confirm I can again get the warning with the suggestion above. Thank you!
